exporter/splunkhecexporter/heartbeat.go

```diff
@@ -0,0 +1,154 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/splunkhecexporter"
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"sync"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type heartbeater struct {
+	config     *Config
+	pushLogFn  func(ctx context.Context, ld plog.Logs) error
+	hbRunOnce  sync.Once
+	hbDoneChan chan struct{}
+
+	// Observability
+	heartbeatSuccessTotal *stats.Int64Measure
+	heartbeatErrorTotal   *stats.Int64Measure
+	tagMutators           []tag.Mutator
+}
+
+func newHeartbeater(config *Config, pushLogFn func(ctx context.Context, ld plog.Logs) error) *heartbeater {
+	prefix := config.HecHeartbeat.ObservabilityName
+	dimensions := config.HecHeartbeat.ObservabilityDimensions
+	tags := []tag.Key{}
+	tagMutators := []tag.Mutator{}
+	for key, val := range dimensions {
+		newTag := tag.MustNewKey(key)
+		tags = append(tags, newTag)
+		tagMutators = append(tagMutators, tag.Insert(newTag, val))
+	}
+
+	heartbeatSuccessTotal := stats.Int64(
+		fmt.Sprintf("%s_success_total", prefix),
+		"number of heartbeats made to the destination",
+		stats.UnitDimensionless)
+
+	heartbeatSuccessTotalView := &view.View{
+		Name:        heartbeatSuccessTotal.Name(),
+		Description: heartbeatSuccessTotal.Description(),
+		TagKeys:     tags,
+		Measure:     heartbeatSuccessTotal,
+		Aggregation: view.Sum(),
+	}
+
+	heartbeatErrorTotal := stats.Int64(
+		fmt.Sprintf("%s_error_total", prefix),
+		"number of heartbeats made to destination that failed",
+		stats.UnitDimensionless)
+
+	heartbeatErrorTotalView := &view.View{
+		Name:        heartbeatErrorTotal.Name(),
+		Description: heartbeatErrorTotal.Description(),
+		TagKeys:     tags,
+		Measure:     heartbeatErrorTotal,
+		Aggregation: view.Sum(),
+	}
+
+	_ = view.Register(heartbeatSuccessTotalView, heartbeatErrorTotalView)
```

Comment:
do you know where this gets exposed? Are you tying in to the collector metrics? I am familiar with https://pkg.go.dev/go.opentelemetry.io/collector/obsreport

Reply:
yeah this is tied with the collector metrics. What our application does is using prometheus receiver and scrape against these metrics.